### PR TITLE
various dev-ey changes

### PR DESCRIPTION
**Pull request by andrewj-brown**
- `black` all python files except ./unimplemented. git hooks et al to be added shortly.
- remove all usage of bot.get_all_channels() in favour of server-based lookup.
- slightly more comprehensible error messages
- net effect of the above two is that the bot no longer uses IDs at all, except for one (1) env variable for its "canonical server id"
- allow committee to bypass the membercount ratelimit via an optional "force" parameter, because discord perms are annoying and restarting the bot every time is annoyinger

### CODE REVIEW

File: uqcsbot/uptime.py
Comment (49Indium) [inline uqcsbot/uptime.py:27]: Really... Fine

**Review by 49Indium — COMMENTED** [review]
That was a lot of changes.

I agree with the linter for most of it. The other changes look good. I'm not knowledgeable enough with github workflows to review `black.yml`, but the rest looks good to me bar comments provided.

File: uqcsbot/uptime.py
Comment (andrewj-brown) [reply uqcsbot/uptime.py:27]: hey, you weren't supposed to notice that

File: uqcsbot/advent.py
Comment (andrewj-brown) [reply uqcsbot/advent.py:433]: modifying the actual functionality of advent is out-of-scope for this PR; the whole cog needs to be rewritten to use slash commands

File: uqcsbot/channels.py
Comment (andrewj-brown) [reply uqcsbot/channels.py:1]: As long as Rachel's committee-are-in-every-channel concern got resolved (I suspect that was a discord thing), the whole cog will just be deleted. Nevertheless, unthinking usage of `black ./uqcsbot/*.py` command formats the file, so no thought required!

File: uqcsbot/haiku.py
Comment (andrewj-brown) [reply uqcsbot/haiku.py:1]: I'll be trying to avoid any conflicts, which means waiting for extant PRs to be merged into main, merging main into this, and rerunning `black ./uqcsbot/*.py` to autoformat them all again.

File: uqcsbot/jobs_bulletin.py
Comment (andrewj-brown) [reply uqcsbot/jobs_bulletin.py:66]: I did consider that but I ceeb'ed the extra logic required. Realistically the channel-not-found-handlers should only be useful for first-time-setup, and it's not hard for a user to see that message and check which channels aren't in their testing server.

Still, I can add it if you want...

File: uqcsbot/member_counter.py
Comment (andrewj-brown) [reply uqcsbot/member_counter.py:62]: That just means if a user doesn't have the manage_guild permission, the force arg is silently ignored. I think this is a better case than trying to tell people "you're not allowed to do that" - *someone* will be a twit with it, easier for the command to just not provide any feedback when it doesn't need to.

File: uqcsbot/starboard.py
Comment (andrewj-brown) [reply uqcsbot/starboard.py:238]: That stuff doesn't actually occur in SB2.0, thanks to a bit of a logic rework, and I'd like that to be merged prior to #94 and styled then.

File: uqcsbot/text.py
Comment (andrewj-brown) [reply uqcsbot/text.py:94]: Cheers for pointing this out. `black`'s rules for formatting itself are consistent, but it allows some leeway in manual formatting to keep under line-length limits, so I've cut them back to 14 per line.

File: uqcsbot/text.py
Comment (andrewj-brown) [reply uqcsbot/text.py:196]: Same as above, but less items per line.

**Comment by andrewj-brown** [comment]
> That was a lot of changes.
> 
> I agree with the linter for most of it. The other changes look good. I'm not knowledgeable enough with github workflows to review `black.yml`, but the rest looks good to me bar comments provided.

`black.yml` just describes a workflow run for github to run the autoformatter in "check mode" as a test for all PRs. In future, once this PR is merged, we'll be expecting people's code to be style-conformant with `black`, with absolutely minimal usage of `# fmt: off` and the like to bypass.

File: uqcsbot/haiku.py
Comment (49Indium) [reply uqcsbot/haiku.py:1]: That still might be a few weeks away, so I'd merge this in first, and just deal with the ugly lists for a few weeks

File: uqcsbot/jobs_bulletin.py
Comment (49Indium) [reply uqcsbot/jobs_bulletin.py:69]: I think it would be good for set up reasons to be able to see which channels were found

**Review by 49Indium — APPROVED** [review]
LGTM

**Review by JamesDearlove — APPROVED** [review]
lgtm

**Pull request merged**
